be2-scala/src/test/scala/ch/epfl/pop/decentralized/MonitorSuite.scala

```diff
@@ -23,58 +27,76 @@ class MonitorSuite extends TestKit(ActorSystem("MonitorSuiteActorSystem")) with
     val monitorRef = system.actorOf(
-      Monitor.props(testProbe.ref, heartbeatRate = 2.seconds, messageDelay = 1.seconds)
+      Monitor.props(testProbe.ref, heartbeatRate = fastRate, messageDelay = fastRate)
     )
 
     testProbe.send(monitorRef, Monitor.AtLeastOneServerConnected)
-    testProbe.expectMsgType[Monitor.GenerateAndSendHeartbeat]
-    testProbe.expectMsgType[Monitor.GenerateAndSendHeartbeat]
-    testProbe.expectMsgType[Monitor.GenerateAndSendHeartbeat]
+    testProbe.expectMsgType[Monitor.GenerateAndSendHeartbeat](timeout)
+    testProbe.expectMsgType[Monitor.GenerateAndSendHeartbeat](timeout)
+    testProbe.expectMsgType[Monitor.GenerateAndSendHeartbeat](timeout)
   }
 
   test("monitor should schedule single heartbeat when receiving a Right GraphMessage") {
 
+    val testProbe = TestProbe()
+    val monitorRef = system.actorOf(
+      Monitor.props(testProbe.ref, heartbeatRate = slowRate, messageDelay = fastRate)
+    )
+
+    // Needed to tell monitor ConnectionMediatorRef
+    val sink = Monitor.sink(monitorRef)
+    testProbe.send(monitorRef, Monitor.AtLeastOneServerConnected)
+
+    Source.single(Right(JsonRpcRequestExample.subscribeRpcRequest)).to(sink).run()
+    testProbe.expectMsgType[Monitor.GenerateAndSendHeartbeat](timeout)
+  }
+
+  test("monitor should not schedule single hearbeats when receiving a heartbeat or get_messages_by_id") {
+
     val heartbeat = JsonRpcRequest(RpcValidator.JSON_RPC_VERSION, MethodType.HEARTBEAT, new ParamsWithMap(Map.empty), None)
     val getMessagesById = JsonRpcRequest(RpcValidator.JSON_RPC_VERSION, MethodType.GET_MESSAGES_BY_ID, new ParamsWithMap(Map.empty), None)
 
     val testProbe = TestProbe()
     val monitorRef = system.actorOf(
-      Monitor.props(testProbe.ref, heartbeatRate = 60.seconds, messageDelay = 1.seconds)
+      Monitor.props(testProbe.ref, heartbeatRate = slowRate, messageDelay = fastRate)
     )
 
     // Needed to tell monitor ConnectionMediatorRef
     val sink = Monitor.sink(monitorRef)
     testProbe.send(monitorRef, Monitor.AtLeastOneServerConnected)
-
     Source.single(Right(heartbeat)).to(sink).run()
-    testProbe.expectNoMessage(5.seconds)
 
+    testProbe.expectNoMessage(timeout)
+
+    // Check that monitor is still doing fine when receiving other Right(...)
     Source.single(Right(JsonRpcRequestExample.subscribeRpcRequest)).to(sink).run()
-    testProbe.expectMsgType[Monitor.GenerateAndSendHeartbeat]
+    testProbe.expectMsgType[Monitor.GenerateAndSendHeartbeat](timeout)
 
     Source.single(Right(getMessagesById)).to(sink).run()
-    testProbe.expectNoMessage(5.seconds)
+    testProbe.expectNoMessage(timeout)
   }
 
   test("monitor should send heartbeats only when servers are connected") {
 
     val testProbe = TestProbe()
     val monitorRef = system.actorOf(
-      Monitor.props(testProbe.ref, heartbeatRate = 1.seconds, messageDelay = 60.seconds)
+      Monitor.props(testProbe.ref, heartbeatRate = fastRate, messageDelay = slowRate)
```

Comment:
`messageDelay` should be `fastRate` otherwise even if an unexpected single HB is scheduled it would not be sent before the test timeout

Reply:
Good point ! The test wasn't very insightful that way...